pkg/client/api.go

```diff
@@ -53,13 +54,26 @@ type httpAPIClient struct {
 func (c *httpAPIClient) Do(ctx context.Context, verb, endpoint string, query url.Values) (APIResponse, error) {
 	u := *c.baseURL
 	u.Path = path.Join(c.baseURL.Path, endpoint)
-	u.RawQuery = query.Encode()
-	req, err := http.NewRequest(verb, u.String(), nil)
+	var reqBody io.Reader
+	if verb == http.MethodGet {
+		u.RawQuery = query.Encode()
+	} else if verb == http.MethodPost {
+		reqBody = strings.NewReader(query.Encode())
+	}
+
+	req, err := http.NewRequest(verb, u.String(), reqBody)
 	if err != nil {
 		return APIResponse{}, fmt.Errorf("error constructing HTTP request to Prometheus: %v", err)
 	}
 	req.WithContext(ctx)
-	req.Header = c.headers
+	for key, values := range c.headers {
+		for _, value := range values {
+			req.Header.Add(key, value)
+		}
+	}
```

Comment:
what is the reasoning behind this change? It shouldn't be needed

Reply:
In my understanding, the original code set the reference of the singleton http headers object to `req.Header`, then `req.Header.Set("Content-Type", "application/x-www-form-urlencoded")` modified the singleton object. Therefore all subsequent http reqeust has `Content-Type: application/x-www-form-urlencoded` regardless of the http method.
Is the behavior no problem?